File: LeetCode/1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.cpp
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        int s1 = str1.length(), s2 = str2.length(), gcd = __gcd(s1, s2);
        string prefix = str1.substr(0, gcd);
        for (int i = 0; i < s2; i += gcd) {
            if (str2.substr(i, gcd) != prefix)
                return "";
        }
        for (int i = gcd; i < s1; i += gcd) {
            if (str1.substr(i, gcd) != prefix)
                return "";
        }
        return prefix;
    }
};

File: LeetCode/1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.py
from math import gcd

class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        GCD = gcd(len(str1), len(str2))
        prefix = str1[:GCD]
        
        for i in range(0, len(str1), GCD):
            if str1[i : i + GCD] != prefix:
                return ""
        for i in range(0, len(str2), gcd):
            if str2[i : i + gcd] != prefix:
                return ""

        return prefix
